src/sage/rings/polynomial/laurent_polynomial_mpair.pyx

```diff
@@ -1183,6 +1183,61 @@ cdef class LaurentPolynomial_mpair(LaurentPolynomial):
             raise TypeError("x must be a generator of parent")
         return self._poly.degree(self._parent._R.gens()[i]) + self._mon[i]
 
+    def valuation(self, x=None):
+        """
+        Return the valuation of ``x`` in ``self``.
+
+        If ``x`` is ``None``, return the minimal valuation of ``self``.
+
+        EXAMPLES::
+
+            sage: R.<x,y> = LaurentPolynomialRing(ZZ)
+            sage: f = 2*x^2*y^-3 - 13*x^-1*y^-3 + 2*x^2*y^-5 - 2*x^-3*y^2
+            sage: f.valuation()
+            -4
+            sage: f.valuation(x)
+            -3
+            sage: f.valuation(y)
+            -5
+            sage: R.zero().valuation()
+            +Infinity
+
+        TESTS:
+
+        If supplied, ``x`` must be a generator::
+
+            sage: R.<x,y> = LaurentPolynomialRing(ZZ)
+            sage: f = 1 + x + x^2*y^-1
+            sage: f.valuation(1)
+            Traceback (most recent call last):
+            ...
+            TypeError: x must be a generator of parent
+        """
+        # Valuation of zero polynomial is defined to be +Infinity
+        if self.is_zero():
+            return Infinity
+
+        # When x is None find the minimal valuation by finding the minimal
+        # valuation of the sum of exponents
+        if x is None:
+            val = min(sum(e) for e in self.exponents())
+            return val
+
+        # Get the index of the gen
+        cdef tuple g = <tuple > self._parent.gens()
+        cdef Py_ssize_t i
+        cdef bint no_generator_found = True
+        for i in range(len(g)):
+            if g[i] == x:
+                no_generator_found = False
+                break
+        if no_generator_found:
+            raise TypeError("x must be a generator of parent")
```

Comment:
```suggestion
        try:
            I = g.index(x)
        except ValueError:  # not in the tuple
            raise TypeError(f"{x} is not a generator")
```
This can be simplified (and is likely faster).